cargo-process.el

```diff
@@ -336,6 +349,24 @@ Without the prefix argument, return DEFAULT immediately."
       (read-shell-command "Cargo command: " default)
     default))
 
+(defun cargo-process--get-dependencies (manifest)
```

Comment:
I wouldn't mind making `manifest` optional and defaulting to `(concat (cargo-process--project-root) "Cargo.toml"))`. This would remove some boilerplate.

Reply:
Had to do a lot of poking on this function, because there are quite a lot of issues (not trying to be mean!). But I ended up making the adjustments. Not sure if `'sexp` will cause issues or not, but right now it's a better alternative to `'word`

``` Emacs Lisp
(defun cargo-process--get-dependencies (&optional manifest)
  "Extract the list of dependencies from the
MANIFEST (i.e. Cargo.toml)."
  (with-current-buffer (find-file-noselect (or manifest
                                               (cargo-process--project-root "Cargo.toml")))
    (save-excursion
      (let ((deps-list))
        (widen)
        (goto-char (point-min))
        (search-forward "[dependencies]" nil t)
        (while (re-search-forward "^[a-zA-Z\-\_]* *=" nil t)
          (beginning-of-line)
          (setq deps-list
                (cons (thing-at-point 'sexp t) deps-list))
          (end-of-line))
        deps-list))))
```

This also requires a small change in `cargo-process--project-root`

``` Emacs Lisp
(defun cargo-process--project-root (&optional extra)
  "Find the root of the current Cargo project."
  (let ((root (locate-dominating-file (or buffer-file-name default-directory) "Cargo.toml")))
    (and root (file-truename (concat root extra)))))
```